ast/src/parsed/mod.rs

```diff
@@ -738,6 +738,30 @@ impl<Ref> From<u32> for Expression<Ref> {
         BigUint::from(value).into()
     }
 }
+
+impl<T: FieldElement> From<AlgebraicExpression<T>> for Expression<Reference> {
```

Comment:
There is `pil-analyzer/src/epressionizer.rs` that does this on a large scale. Maybe call that from pilopt instead of adding this here?

Reply:
I knew we had this somewhere! Thanks! I added this just to make it compile